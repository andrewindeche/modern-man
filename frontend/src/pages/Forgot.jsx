import React from 'react';
import { Link } from 'react-router-dom';
import {
  faHome,
} from '@fortawesome/free-solid-svg-icons';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';

const Forgot = () => (
  <div className="forgotPasswordForm">
    <Link to="/">
      <FontAwesomeIcon icon={faHome} className="home-icon" />
      <span className="tooltip-text">Go To Home</span>
    </Link>
    <form>
      <p>Forgot Password? Generate New password</p>
      <span className="nameLabel">
        <div className="input-group" id="forgot-input">
          <label htmlFor="fullName">Enter Your Name</label>
          <input type="text" id="fullName" name="fullName" placeholder="Enter your Name" required />
          <label htmlFor="password">Enter Your Password</label>
          <input type="password" id="password" name="password" placeholder="Enter New Password" required />
        </div>
      </span>
      <button id="forgotButton" type="submit">Reset Password</button>
    </form>
  </div>
);
export default Forgot;
