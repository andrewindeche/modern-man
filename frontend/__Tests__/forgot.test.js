import React from 'react';
import { render } from '@testing-library/react';
import { BrowserRouter as Router } from 'react-router-dom';
import '@testing-library/jest-dom/extend-expect'; 
import Forgot from 'pages/Forgot.jsx';

describe('Forgot component', () => {
  it('renders without crashing', () => {
    const { getByText } = render(
      <Router>
        <Forgot />
      </Router>
    );
    expect(getByText(/Forgot Password?/i)).toBeInTheDocument();
  });
});
