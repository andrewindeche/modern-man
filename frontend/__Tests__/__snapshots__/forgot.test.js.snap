// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`forgot page renders the forgot page 1`] = `
<div
  className="forgotPasswordForm"
>
  <form>
    <p>
      Forgot Password? Generate New password
    </p>
    <span
      className="nameLabel"
    >
      <div
        className="input-group"
      >
        <label>
          Enter Your Name
        </label>
        <input
          name="fullName"
          placeholder="Enter your Name"
          required={true}
          type="text"
        />
      </div>
      <div
        className="input-group"
      >
        <label>
          Enter Your Password
        </label>
        <input
          name="fullName"
          placeholder="Enter New Password"
          required={true}
          type="text"
        />
      </div>
    </span>
    <button
      type="submit"
    >
      Reset Password
    </button>
  </form>
</div>
`;
